plugins/postgres/dbt/adapters/postgres/connections.py

```diff
@@ -131,23 +148,23 @@ def open(cls, connection):
                 connect_timeout=credentials.connect_timeout,
                 **kwargs,
             )
-
             if credentials.role:
                 handle.cursor().execute("set role {}".format(credentials.role))
-
-            connection.handle = handle
-            connection.state = "open"
-        except psycopg2.Error as e:
-            logger.debug(
-                "Got an error when attempting to open a postgres " "connection: '{}'".format(e)
-            )
-
-            connection.handle = None
-            connection.state = "fail"
-
-            raise dbt.exceptions.FailedToConnectException(str(e))
-
-        return connection
+            return handle
+
+        retryable_exceptions = [
+            # OperationalError is subclassed by all psycopg2 Connection Exceptions
+            # and it's raised by generic connection errors without an error code.
+            psycopg2.errors.OperationalError,
+        ]
+
+        return cls.retry_connection(
+            connection,
+            connect=connect,
+            logger=logger,
+            retry_limit=credentials.retries,
```

Comment:
RE: your conversation with @jtcohen6, is credentials the place we've decided to expose this value?

Reply:
I proposed 3 potential places where we could expose the `retries` configuration: https://github.com/dbt-labs/dbt-core/pull/5432#discussion_r920320624

@jtcohen6 felt inclined towards going with the third option (and I agreed):

> 3. In each individual target, which means in each adapters credentials (next to other connection options like `password`, that's how it's done in this PR currently).

The linked discussion has more details about our discussion.